File: lista2/python/main.py
from GF import GF

def main():
    a = GF(70)
    b = GF(1234588)
    c = GF(-30)
    result = GF()

    print("characteristic:", result.characteristic())

    print("a =", a)
    print("b =", b)
    print("c =", c)

    print("b inversed =", b.inverse())

    result = a + b
    print("a + b =", result)

    result = a - b
    print("a - b =", result)

    result = a * b
    print("a * b =", result)

    result = a / b
    print("a / b =", result)

    print("Is a == b?", a == b)
    print("Is a != b?", a != b)
    print("Is a < b?", a < b)
    print("Is a > b?", a > b)

if __name__ == "__main__":
    main()

File: lista2/java/Main.java
public class Main {
    public static void main(String[] args) {
        GF a = new GF(70);
        GF b = new GF(1234588);
        GF c = new GF(-30);
        GF result;

        System.out.println("characteristic: " + a.characteristic());

        System.out.println("a = " + a);
        System.out.println("b = " + b);
        System.out.println("c = " + c);

        System.out.println("b inversed = " + b.inverse());

        result = GF.add(a, b);
        System.out.println("a + b = " + result);

        result = GF.subtract(a, b);
        System.out.println("a - b = " + result);

        result = GF.multiply(a, b);
        System.out.println("a * b = " + result);

        result = GF.divide(a, b);
        System.out.println("a / b = " + result);

        System.out.println("Is a == b? " + (a.equals(b)));
        System.out.println("Is a != b? " + (a.notEquals(b)));
        System.out.println("Is a < b? " + (a.lessThan(b)));
        System.out.println("Is a > b? " + (a.greaterThan(b)));
    }
}
